From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Todd Kjos <tkjos@google.com>
Date: Tue, 17 Sep 2019 14:36:04 -0700
Subject: ANDROID: init: GKI: enable hidden configs for DRM

Add hidden configs to GKI_HACKS_TO_FIX so they
are enabled for loadable DRM modules built
out-of-tree.

Bug: 141266428
Change-Id: Ibeda203afa54edc4f9a1daec9b4e8a928eb934fb
Signed-off-by: Todd Kjos <tkjos@google.com>
---
 init/Kconfig.gki | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/init/Kconfig.gki b/init/Kconfig.gki
index fafcd618037d..6c3f89bd8e25 100644
--- a/init/Kconfig.gki
+++ b/init/Kconfig.gki
@@ -1,3 +1,14 @@
+config GKI_HIDDEN_DRM_CONFIGS
+	bool "Hidden DRM configs needed for GKI"
+	select DRM_KMS_HELPER if (HAS_IOMEM && DRM)
+	select DRM_GEM_CMA_HELPER
+	select DRM_KMS_CMA_HELPER
+	select DRM_MIPI_DSI
+	help
+	  Dummy config option used to enable hidden DRM configs.
+	  These are normally selected implicitely when including a
+	  DRM module, but for GKI, the modules are built out-of-tree.
+
 # Atrocities needed for
 # a) building GKI modules in separate tree, or
 # b) building drivers that are not modularizable
@@ -7,6 +18,7 @@
 #
 config GKI_HACKS_TO_FIX
 	bool "GKI Dummy config options"
+	select GKI_HIDDEN_DRM_CONFIGS
 	help
 	  Dummy config option used to enable core functionality used by
 	  modules that may not be selectable in this config.
